modules/reindex/src/test/java/org/elasticsearch/index/reindex/CancelTests.java

```diff
@@ -85,8 +82,8 @@ public static void clearAllowedOperations() {
     private void testCancel(String action, AbstractBulkByScrollRequestBuilder<?, ?> builder, CancelAssertion assertion) throws Exception {
         createIndex(INDEX);
 
-        // Total number of documents created for this test (~10 per primary shard)
-        int numDocs = getNumShards(INDEX).numPrimaries * 10;
+        // Total number of documents created for this test (~10 per primary shard per worker)
+        int numDocs = getNumShards(INDEX).numPrimaries * 10 * builder.request().getWorkers();
```

Comment:
@tlrx this is fairly fiddly! Can you make sure I didn't make a total mess of it?